client.go

```diff
@@ -753,3 +763,22 @@ func (s *Client) Probe(ctx context.Context, req *ProbeRequest) error {
 		req.RouteHints,
 	)
 }
+
+// AbandonSwap sends the hash of the swap to abandon to all active swaps. Only
+// the addressed swap will abandon itself.
+func (s *Client) AbandonSwap(ctx context.Context,
+	req *AbandonSwapRequest) error {
+
+	if req == nil {
+		return errors.New("no request provided")
+	}
+
+	select {
+	case s.abandonChans[req.SwapHash] <- struct{}{}:
```

Comment:
Access here should also be guarded by the mutex.

Reply:
There's also an issue where if the abandon chan is not read immediately then this call might block and hold the mutex. A buffered channel can help here with the addition of using a non-blocking write:
```

s.executor.Lock()
defer s.executor.Unlock()

select {
    case s.abandonChans[req.SwapHash] <- struct{}{}:
    case <-ctx.Done():
                   return ctx.Err()
    default:
         // This is to avoid writing to a full channel.
    }
```

Reply:
Thanks for the explanation. I've also changed the channel in the resume-swaps case to a buffered one.